react/index.d.ts

```diff
@@ -14,7 +14,7 @@ declare namespace React {
 
     type ReactType = string | ComponentClass<any> | StatelessComponent<any>;
 
-    type Key = string | number;
+    type Key = string | number | null;
```

Comment:
You can use `  React.createElement`   or other methods that created elements, the return type should reflect the actual possible types a variable can actually become.

So if you do something with the returned payload it warns you that the value can actually be `  null`  . I don't actually get the question here since the `  Key`   value can actually be null from the built in methods, regardless of necessity, this reflects the actual React API workings.